src/lfwatch_win32.cpp

```diff
@@ -73,25 +73,32 @@ void emit_events(WatchData &watch){
 	do {
 		info = reinterpret_cast<PFILE_NOTIFY_INFORMATION>(&watch.info_buf[0] + offset);
 		offset += info->NextEntryOffset;
-		//FileNameLength is size in bytes of the 16-bit Unicode string so, compute
-		//the max number of chars that it could contain
-		//This is done to put the null terminator on the end of the string since
-		//Win32 doesn't null-terminate
-		int n_chars = info->FileNameLength / 2;
-		std::vector<wchar_t> wfname(n_chars + 1);
-		std::memcpy(&wfname[0], info->FileName, info->FileNameLength);
-		char fname[MAX_PATH + 1] = { 0 };
-		std::wcstombs(fname, &wfname[0], MAX_PATH);
+		
 		//Since FILE_NOTIFY_CHANGE_FILE_NAME gives all create/delete/rename events it's possible that
 		//we only want create but have gotten one of the other two, so make sure we actually want this
 		uint32_t action = remap_file_action(info->Action);
 		if (action & watch.filter){
+			//FileNameLength is size in bytes of the 16-bit Unicode string so, compute
+			//the max number of chars that it could contain
+			//This is done to put the null terminator on the end of the string since
+			//Win32 doesn't null-terminate
+			int n_chars = info->FileNameLength / 2;
+			std::vector<wchar_t> wfname(n_chars + 1);
+			std::memcpy(&wfname[0], info->FileName, info->FileNameLength);
+			char fname[MAX_PATH + 1];
```

Comment:
Can we also keep the 0 initialization here? I think the null terminator shouldn't be lost since it's added on line 86 since we make `wfname` 1 character bigger, but probably best to keep this 0 initialized.

Reply:
You can keep, makes no harm, but the wcstombs_s guarantees that it will put a \0 at the string end, thats why I removed the 0 initialization. 

I am not a big fan of all those changes that microsoft did to the standard library. Another solution is to modify the cmake file to include the macro _CRT_SECURE_NO_WARNINGS and keep the code unchanged. 

What do you think?

Reply:
Lets keep it as you have in the PR then, since it'll always add the null terminator